src/comparison.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * @license Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ */
+
+export function compareNullable<T>(a: T | undefined, b: T | undefined, compute: (t: T | undefined) => T): -1 | 0 | 1 {
+    if (a === b) {
```

Comment:
if a == null and b === undefined this will return false, if you care

Reply:
yup!